example/string_generator_test.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/python
+"""Test the string generator class."""
+
+import os
+import cherrypy
+from cherrypy.test import helper
+from example.string_generator import StringGeneratorWebService, STRING_FILE
+
+
+class TestStringGenerator(helper.CPWebCase):
+    """The test framework for string generator service."""
+
+    # pylint: disable=no-self-use
+    def setUp(self):
+        """Setup each test by starting the CherryPy server."""
+        self.PORT = 1234
+        self.HOST = '127.0.0.1'
+        if not os.access(STRING_FILE, os.F_OK):
+            os.ftruncate(os.open(STRING_FILE, os.O_WRONLY | os.O_CREAT), 0)
```

Comment:
I believe, `os.open()` may leave unclosed file handler somewhere in memory. Please use `with` to avoid this.